snuba/datasets/discover.py

```diff
@@ -0,0 +1,236 @@
+            ('exception_stacks.mechanism_type', Nullable(String())),
+            ('exception_stacks.mechanism_handled', Nullable(UInt(8))),
+            ('exception_frames.abs_path', Nullable(String())),
+            ('exception_frames.filename', Nullable(String())),
+            ('exception_frames.package', Nullable(String())),
+            ('exception_frames.module', Nullable(String())),
+            ('exception_frames.function', Nullable(String())),
+            ('exception_frames.in_app', Nullable(UInt(8))),
+            ('exception_frames.colno', Nullable(UInt(32))),
+            ('exception_frames.lineno', Nullable(UInt(32))),
+            ('exception_frames.stack_level', Nullable(UInt(16))),
+            ('modules.name', String()),
+            ('modules.version', String()),
+        ])
+
+    def get_transactions_only_columns(self) -> ColumnSet:
+        return ColumnSet([
+            ('trace_id', Nullable(UUID())),
+            ('span_id', Nullable(UInt(64))),
+            ('transaction_name', String()),
+            ('transaction_hash', Nullable(UInt(64))),
+            ('transaction_op', Nullable(String())),
+            ('start_ts', Nullable(DateTime())),
+            ('start_ms', Nullable(UInt(16))),
+            ('finish_ts', Nullable(DateTime())),
+            ('finish_ms', Nullable(UInt(16))),
+            ('duration', Nullable(UInt(32))),
+        ])
+
+
+class DiscoverDataset(TimeSeriesDataset):
+    """
+    Experimental dataset for Discover
+    that coerces the columns of Events and Transactions into the same format
+    and sends a query to either one.
+
+    Currently does this by switching between events and transactions tables
+    depending on the conditions in the provided query.
+    """
+
+    def __init__(self) -> None:
+        super().__init__(
+            dataset_schemas=DatasetSchemas(
+                read_schema=DiscoverSchema(),
+                write_schema=None,
+            ),
+            time_group_columns={},
+            time_parse_columns=['timestamp', 'start_ts', 'finish_ts'],
+        )
+
+    def get_query_processors(self) -> Sequence[QueryProcessor]:
+        return [
+            DiscoverProcessor(),
+        ]
+
+    def get_extensions(self) -> Mapping[str, QueryExtension]:
+        return {
+            'project': ProjectExtension(
+                processor=ProjectWithGroupsProcessor(project_column='project_id')
+            ),
+            'timeseries': TimeSeriesExtension(
+                default_granularity=3600,
+                default_window=timedelta(days=5),
+                timestamp_column='timestamp',
+            ),
+        }
+
+    def column_expr(self, column_name, query: Query, parsing_context: ParsingContext):
+        detected_dataset = detect_dataset(query)
+
+        if detected_dataset == TRANSACTIONS:
+            if column_name == 'type':
+                return "'transaction'"
+            if column_name == 'timestamp':
```

Comment:
There is also the time group alias that is used in timeseries results. It currently maps to the `bucketed_end` alias or `time` for events.

Reply:
I added `time` and `bucketed_end` to the time columns. It would be nice if we could just use `time` for both, but this might require some larger changes.

Reply:
I'm :+1: in using `time` as the standard alias across both datasets.